File: Backend/AstroGame.Shared/Models/Stellar/StellarObjects/Moon.cs
using AstroGame.Shared.Models.Resources;
using AstroGame.Shared.Models.Stellar.BaseTypes;
using AstroGame.Shared.Models.Stellar.Interfaces;
using System.Collections.Generic;
using AstroGame.Shared.Enums;

namespace AstroGame.Shared.Models.Stellar.StellarObjects
{
    public class Moon : StellarObject, IRenderable
    {
        public Moon()
        {
        }

        public Moon(StellarSystem parentSystem) : base(parentSystem)
        {
        }

        public double Scale { get; set; }
        public double AxialTilt { get; set; }
        public List<StellarObjectResource> Resources { get; set; }
        public override StellarObjectType StellarObjectType => StellarObjectType.Moon;
    }
}

File: Frontend/apps/astrogame_app/lib/models/stellar/stellar_objects/moon.dart
import 'package:astrogame_app/communications/converters/guid_converter.dart';
import 'package:astrogame_app/models/common/coordinates.dart';
import 'package:astrogame_app/models/resources/stellar_object_resource.dart';
import 'package:astrogame_app/models/stellar/base_types/stellar_object.dart';
import 'package:json_annotation/json_annotation.dart';

part 'moon.g.dart';

@GuidConverter()
@JsonSerializable()
class Moon extends StellarObject {
  @JsonKey()
  late double scale;

  @JsonKey()
  late double axialTilt;

  @JsonKey()
  List<StellarObjectResource> resources = [];

  @JsonKey()
  late int order;

  Moon();

  factory Moon.fromJson(Map<String, dynamic> json) => _$MoonFromJson(json);
  Map<String, dynamic> toJson() => _$MoonToJson(this);
}
